src/cmap/connection_pool.ts

```diff
@@ -355,6 +356,7 @@ export class ConnectionPool extends TypedEventEmitter<ConnectionPoolEvents> {
    * explicitly destroyed by the new owner.
    */
   async checkOut(): Promise<Connection> {
+    this.checkOutTime = Date.now();
```

Comment:
How does this work when multiple connections are checked out of the pool concurrently?  Each checkout request would override the previous request's checkout time.

Reply:
I considered this as well, but assumed we don't checkout multiple connections simultaneously. I'll store `checkoutTime` on the `Connection` class directly not on the pool, how does that sound?